File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> set=new HashSet<>();
        for(int i:nums)set.add(i);
        int ans=0;
        for(int i:set){
            if(!set.contains(i-1)){
                int num=i+1;
                while(set.contains(num)){
                    num++;
                }
                if(num-i>ans)ans=num-i;
            }
        }
        return ans;
    }
}

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s=set()
        for i in nums:
            s.add(i)
        ans=0
        for i in s:
            if i-1 not in s:
                num=i
                while(num in s):
                    num+=1
                if num-i>ans:
                    ans=num-i
        return ans